src/btree/bt_page.c

```diff
@@ -632,6 +632,18 @@ __inmem_row_leaf(WT_SESSION_IMPL *session, WT_PAGE *page)
             tombstone->prepare_state = WT_PREPARE_INPROGRESS;
             F_SET(tombstone, WT_UPDATE_PREPARE_RESTORED_FROM_DS);
             F_SET(upd, WT_UPDATE_RESTORED_FROM_DS);
+
+            /*
+             * If the update and tombstone both are from the same transaction, set the update also
+             * as prepare in-progress.
+             */
+            if (upd->durable_ts == tombstone->durable_ts && upd->start_ts == tombstone->start_ts &&
+              upd->txnid == tombstone->txnid) {
```

Comment:
Just a miner point. Can we compare txnid before timestamps?

Reply:
Fixed.